### PR TITLE
Support Gson Plugin

### PR DESCRIPTION
**Pull request by WithLin**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

**Comment by JaredTan95** [comment]
Cool, it's 1st Serialization component plugin, right? It's better to put it in the default plugin dir.

**Comment by ascrutae** [comment]
I'm afraid not that I don't agree with put GSON plugin into the default plugins.  Because of GSON is a serialization component, that is mean some frameworks maybe use it,  If so, The application that use those framework will generate some useless traces about GSON, I think this maybe waste the network resource.   I prefer put it in optional plugin to put it in default plugin

**Comment by wu-sheng** [comment]
@ascrutae but, even in optional, your scenario still exists. For auto instrument, you can't choose which parts are going to generate span, right? The user don't know the framework will use it or not. So, document can't help in this way either, because you can't give detail suggestion.

Even the framework uses this, do you think we run slower or similar than gson? I assume we are much faster than it.

**Comment by ascrutae** [comment]
I mean that we COULD NOT generate those tracer to resolve some network resource in default . If someone want to trace GSON, that' ok, put this plugin into the default dir and it will works.

**Comment by wu-sheng** [comment]
What do you mean, generate those tracer?

Guarantee? Or ?

**Comment by candyleer** [comment]
I am agree with @ascrutae 's opinion,  `Gson` is a serilization library,It may be used by some framework and it is not `network component`,the default dir maybe should put some `network component` I think,If some one want to trace `serilization` ,he can copy this plugin from optional to default

**Comment by wu-sheng** [comment]
Look like we have 2 for default and 2 for optional. Then for safety, let's move it to optional now. @WithLin Also, this movement means, you need to add a section at here

https://github.com/apache/incubator-skywalking/tree/master/docs/en/setup/service-agent/java-agent#plugins

And a linked document to describe what this plugin for.

**Comment by WithLin** [comment]
i am already  fix it  in  test-case in  repo. it is  expected data problem.

File: docs/en/setup/service-agent/java-agent/Supported-list.md [hunk 1]
```diff
@@ -56,6 +56,8 @@
   * [Elastic Job](https://github.com/elasticjob/elastic-job) 2.x
 * OpenTracing community supported
 * [Canal: Alibaba mysql database binlog incremental subscription & consumer components](https://github.com/alibaba/canal) 1.0.25 -> 1.1.2
+* JSON
+  * [GSON](https://github.com/google/gson) 2.8.x (Optional¹)
```
Comment (wu-sheng) [inline docs/en/setup/service-agent/java-agent/Supported-list.md]: This is optional 2

Comment (WithLin) [reply docs/en/setup/service-agent/java-agent/Supported-list.md]: Done.

**Pull request merged**